### PR TITLE
Sort default Groups and Admins

### PR DESCRIPTION
**Pull request by wodka**
Sorts all admins and groups regardless of their order in configuration. (https://github.com/sonata-project/SonataAdminBundle/issues/3018)

* ```<tag name="sonata.admin" group="Z-Group" label="A-Admin"/>```
* ```<tag name="sonata.admin" group="Y-Group" label="C-Admin"/>```
* ```<tag name="sonata.admin" group="Y-Group" label="B-Admin"/>```

would result in the following order:

- Y-Group:
 - B-Admin
 - C-Admin
- Z-Group:
 - A-Admin

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Groups and Admins sorting is a good idea but should be proposed as an option and setted to false as default.

For my case, I manually ordered Admins ans Groups on a logical way and wouldn't see it changed on next update.

Regards.

**Comment by wodka** [comment]
@Soullivaneuh makes sense, I changed it now to not change the default order unless the config setting sonata_admin.options.admin_sort is set to true

File: DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php [hunk 1]
```diff
@@ -95,7 +95,7 @@ public function process(ContainerBuilder $container)
 
                 $groupDefaults[$resolvedGroupName]['items'][] = array(
                     'admin'        => $id,
-                    'label'        => '',
+                    'label'        => !empty($attributes['label'])?$attributes['label']:'',
```
Comment (Soullivaneuh) [inline DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php]: Ternary should be spaced.

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -76,6 +76,7 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->children()
                         ->booleanNode('html5_validate')->defaultValue(true)->end()
+                        ->booleanNode('admin_sort')->defaultValue(false)->end()
```
Comment (Soullivaneuh) [inline DependencyInjection/Configuration.php]: `defaultValue(false)` can be replaced by `defaultFalse()`.

Please see: https://github.com/symfony/Config/blob/2.3/Definition/Builder/NodeDefinition.php#L241

File: Resources/doc/reference/configuration.rst [hunk 3]
```diff
@@ -57,6 +57,7 @@ Full Configuration Options
             title_logo:           bundles/sonataadmin/logo_title.png
             options:
                 html5_validate:       true
+                admin_sort:          false
```
Comment (Soullivaneuh) [inline Resources/doc/reference/configuration.rst]: Will be a little punctilious but should be admin**s**_sort, isn't it? :-)

**Comment by Soullivaneuh** [comment]
@wodka code reviewed.

**Comment by wodka** [comment]
@Soullivaneuh finished the requested changes

File: DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php [hunk 4]
```diff
@@ -140,6 +140,26 @@ public function process(ContainerBuilder $container)
                     $groups[$resolvedGroupName]['roles'] = $groupDefaults[$resolvedGroupName]['roles'];
                 }
             }
+        } elseif($container->getParameter('sonata.admin.configuration.admins_sort')) {
```
Comment (Soullivaneuh) [inline DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php]: Should be `elseif (` according to PSR-2.

**Comment by Soullivaneuh** [comment]
@wodka Missing one sorry. :-)

You can use php-cs-fixer to fix PSR-1/2 standards on your modifications: http://cs.sensiolabs.org/ :+1:

**Comment by Soullivaneuh** [comment]
And at least, if you can squash your commits after that, this will make a better git history. ;-)

**Comment by wodka** [comment]
squashed the commits into one - but I have no idea why it shows now 2 commits... -> If I try to create a new pull request it's only one commit (since the other is from master)

@Soullivaneuh ran psr2 check only on AddDependencyCallsCompilerPass.php -> there would have been more than 70 files changed for the whole project

**Comment by Soullivaneuh** [comment]
> squashed the commits into one - but I have no idea why it shows now 2 commits...

I think you got one commit that was not on this PR when squashing. :-)

> ran psr2 check only on AddDependencyCallsCompilerPass.php

Don't run it for the moment. WIP on #3038.

Well, I think opening a new clean PR will be simpler. Ping me on it when it's done. :+1:

**Comment by wodka** [comment]
create a new pull request in https://github.com/sonata-project/SonataAdminBundle/pull/3041

**Pull request closed without merging**